pkg/api/validation/validation.go

```diff
@@ -1074,14 +1076,12 @@ func validateDNSPolicy(dnsPolicy *api.DNSPolicy) errs.ValidationErrorList {
 	return allErrors
 }
 
-func validateHostNetwork(hostNetwork bool, containers []api.Container) errs.ValidationErrorList {
+func validateHostNetwork(containers []api.Container) errs.ValidationErrorList {
 	allErrors := errs.ValidationErrorList{}
-	if hostNetwork {
-		for _, container := range containers {
-			for _, port := range container.Ports {
-				if port.HostPort != port.ContainerPort {
-					allErrors = append(allErrors, errs.NewFieldInvalid("containerPort", port.ContainerPort, "containerPort must match hostPort if hostNetwork is set to true"))
-				}
+	for _, container := range containers {
```

Comment:
I like this change, but it is unrelated,  Maybe it can be broken to a different PR?